tests/unit/processing/test_spatial_assets_utils.py

```diff
@@ -0,0 +1,315 @@
+        "valid_uns": valid_uns,
+        "valid_spatial_data": valid_spatial_data,
+        "invalid_uns": invalid_uns,
+        "spatial_processor": spatial_processor,
+        "output_folder": output_folder,
+        "cxg_container": cxg_container,
+        "group_metadata_name": group_metadata_name,
+        "ctx": ctx,
+        "mock_spatial_processor": mock_spatial_processor,
+    }
+
+
+def test__valid_input_metadata_copy(setup_spatial_processor):
+    """
+    Test case for verifying the required metadata is present
+    """
+    spatial_processor = setup_spatial_processor["spatial_processor"]
+    valid_spatial_data = setup_spatial_processor["valid_spatial_data"]
+    library_id = setup_spatial_processor["library_id"]
+
+    # Expected output based on the input data
+    expected_output = {
+        library_id: {
+            "images": {"hires": valid_spatial_data["images"]["hires"], "fullres": []},
+            "scalefactors": {
+                "spot_diameter_fullres": valid_spatial_data["scalefactors"]["spot_diameter_fullres"],
+                "tissue_hires_scalef": valid_spatial_data["scalefactors"]["tissue_hires_scalef"],
+            },
+        }
+    }
+
+    result = spatial_processor.filter_spatial_data(valid_spatial_data, library_id)
+
+    assert result == expected_output, f"Expected {expected_output}, but got {result}"
+
+
+def test__invalid_input_metadata_copy(setup_spatial_processor):
+    """
+    Test case to verify that a KeyError is raised when passing input metadata
+    """
+    spatial_processor = setup_spatial_processor["spatial_processor"]
+    invalid_uns = setup_spatial_processor["invalid_uns"]
+    library_id = setup_spatial_processor["library_id"]
+    with pytest.raises(KeyError):
+        spatial_processor.filter_spatial_data(invalid_uns, library_id)
+
+
+def test__full_high_res_image_present(setup_spatial_processor):
+    """
+    Test that the default image is fullres if present otherwise fall back to hires
+    """
+    spatial_processor = setup_spatial_processor["spatial_processor"]
+    valid_spatial_data = setup_spatial_processor["valid_spatial_data"]
+
+    image_array_uint8 = spatial_processor._prepare_image(valid_spatial_data)
+    assert image_array_uint8.shape == (20, 20, 3)
+
+    valid_hires_uns = copy.deepcopy(valid_spatial_data)
+    valid_hires_uns["images"].pop("fullres", None)
+    image_array_uint8 = spatial_processor._prepare_image(valid_hires_uns)
+    assert image_array_uint8.shape == (10, 10, 3)
+
+    non_valid_uns = copy.deepcopy(valid_hires_uns)
+    non_valid_uns["images"].pop("hires", None)
+    with pytest.raises(KeyError):
+        image_array_uint8 = spatial_processor._prepare_image(non_valid_uns)
+
+
+def test__process_and_flip_image(setup_spatial_processor):
+    """
+    Test the image processing method
+    """
+    spatial_processor = setup_spatial_processor["spatial_processor"]
+    test_image_array = (np.random.rand(10, 10, 3) * 255).astype(np.uint8)
```

Comment:
nit: It helps to have non-random values for reproducible tests. Is there a specific reason we are using random? Maybe you could create a pytest fixture with one large ndarray, and take slices of that for testing.

Reply:
I agree, but for now I will leave it with random values - array for full res is huge and can go up to 2 GB in size.